File: S1MCPClient/src/models/request.py
"""Request model for JSON-RPC communication."""

from typing import Any, Dict, Optional
from pydantic import BaseModel, Field


class Request(BaseModel):
    """JSON-RPC request model matching the mod's Request structure."""
    
    id: int = Field(..., description="Unique request identifier")
    method: str = Field(..., description="Method name to invoke")
    params: Optional[Dict[str, Any]] = Field(None, description="Method parameters as a JSON object")
    
    class Config:
        """Pydantic configuration."""
        json_schema_extra = {
            "example": {
                "id": 1,
                "method": "get_npc",
                "params": {
                    "npc_id": "kyle_cooley"
                }
            }
        }

File: S1MCPServer/Models/Request.cs
using System.Text.Json.Serialization;

namespace S1MCPServer.Models;

/// <summary>
/// Represents a JSON-RPC request from the MCP server.
/// </summary>
public class Request
{
    /// <summary>
    /// Unique request identifier.
    /// </summary>
    [JsonPropertyName("id")]
    public int Id { get; set; }

    /// <summary>
    /// Method name to invoke.
    /// </summary>
    [JsonPropertyName("method")]
    public string Method { get; set; } = string.Empty;

    /// <summary>
    /// Method parameters as a JSON object.
    /// </summary>
    [JsonPropertyName("params")]
    public Dictionary<string, object>? Params { get; set; }
}


